server/src/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2608,6 +2609,21 @@ public UserVm destroyVm(DestroyVMCmd cmd) throws ResourceUnavailableException, C
         CallContext ctx = CallContext.current();
         long vmId = cmd.getId();
         boolean expunge = cmd.getExpunge();
+        // check if VM exists
```

Comment:
What is the purpose of these checks? If at all required should be done after validating that caller has proper permission as done immediately after these changes